In [2]:
# Import pandas
import pandas as pd

In [13]:
# Create a DataFrame with balance.txt
df = pd.read_csv("balance.txt", sep=" ")

Write the code needed to produce a report that provides the following information:
* Compare the average income based on ethnicity.
* On average, do married or single people have a higher balance?
* What is the highest income in our dataset?
* What is the lowest income in our dataset?
* How many cards do we have recorded in our dataset? (Hint: use `sum()`)
* How many females do we have information for vs how many males? (Hint: use `count()` for a list of all methods for computation of descriptive stats, explore the [pandas documentation](https://pandas.pydata.org/pandas-docs/stable/reference/frame.html#computations-descriptive-stats)).




In [22]:
print(df.groupby("Ethnicity")["Income"].mean())

Ethnicity
African American    47.682101
Asian               44.187833
Caucasian           44.521945
Name: Income, dtype: float64


It is clear form the above output that African American's have the highest income, followed by Caucasians, and finally Asians

In [ ]:
print(df.groupby("Married")["Balance"].mean())

In [ ]:
So, on average, single people have a higher balance

In [30]:
df.sort_values(by="Income",ascending=False)

Balance   Income  Limit  Rating  Cards  Age  Education  Gender Student  \
28   35.271011  186.634  13414     949      2   41         14  Female      No   
323  33.745580  182.728  13913     982      4   98         17    Male      No   
355  34.034656  180.682  11966     832      2   58          8  Female      No   
261  38.785123  180.379   9310     665      3   67          8  Female     Yes   
275  30.212080  163.329   8732     636      3   50         14    Male      No   
..         ...      ...    ...     ...    ...  ...        ...     ...     ...   
262   7.653979   10.588   4049     296      1   66         13  Female      No   
235   7.503813   10.503   2923     232      3   25         18  Female      No   
199   6.687342   10.403   4159     310      3   43          7    Male      No   
250   8.573448   10.363   2430     191      2   47         18  Female      No   
58    9.180797   10.354   3480     281      2   70         17    Male      No   

    Married         Ethnicity  
28      Yes  African American  
323     Yes         Caucasian  
355     Yes  African American  
261     Yes             Asian  
275     Yes         Caucasian  
..      ...               ...  
262     Yes         Caucasian  
235     Yes  African American  
199     Yes             Asian  
250     Yes             Asian  
58      Yes         Caucasian  

[400 rows x 11 columns]

It follows that row 28 has the highest income and row 58 has the lowest income.

In [32]:
df["Cards"].sum()

1183

We have a total number of 1183 cards in our dataset.

In [44]:
df[df.Gender == "Male"].count()


Balance      193
Income       193
Limit        193
Rating       193
Cards        193
Age          193
Education    193
Gender       193
Student      193
Married      193
Ethnicity    193
dtype: int64

There are 193 males in the dataset. Accordingly, there must be 207 females.